src/calendar/Calendar.ts

```diff
@@ -0,0 +1,391 @@
+	{short: 'Dec', long: 'December'}
+];
+
+export const DEFAULT_WEEKDAYS = [
+	{short: 'Sun', long: 'Sunday'},
+	{short: 'Mon', long: 'Monday'},
+	{short: 'Tue', long: 'Tuesday'},
+	{short: 'Wed', long: 'Wednesday'},
+	{short: 'Thu', long: 'Thursday'},
+	{short: 'Fri', long: 'Friday'},
+	{short: 'Sat', long: 'Saturday'}
+];
+
+export const DEFAULT_LABELS: CalendarMessages = {
+	chooseMonth: 'Choose Month',
+	chooseYear: 'Choose Year',
+	previousMonth: 'Previous Month',
+	nextMonth: 'Next Month'
+};
+
+@theme(css)
+export default class Calendar extends ThemeableMixin(WidgetBase)<CalendarProperties> {
+	private _callDateFocus = false;
+	private _defaultDate = new Date();
+	private _focusedDay = 1;
+	private _monthLabelId = uuid();
+	private _monthPopupOpen = false;
+	private _registry: WidgetRegistry;
+
+	protected onElementUpdated(element: HTMLElement, key: string) {
+		if (this._callDateFocus) {
+			const { month } = this._getMonthYear();
+			// focus date with correct key
+			if (key === (this._focusedDay + '-' + month)) {
+				element.focus();
+				this._callDateFocus = false;
+			}
+		}
+	}
+
+	@onPropertiesChanged()
+	protected onPropertiesChanged(evt: PropertiesChangeEvent<this, CalendarProperties>) {
+		const { customDateCell = CalendarCell } = this.properties;
+
+		// update custom option registry
+		if ( !this._registry || includes(evt.changedPropertyKeys, 'customDateCell')) {
+			const registry = this._createRegistry(customDateCell);
+
+			if (this._registry) {
+				this.registries.replace(this._registry, registry);
+			}
+			else {
+				this.registries.add(registry);
+			}
+			this._registry = registry;
+		}
+	}
+
+	private _createRegistry(customDateCell: any) {
+		const registry = new WidgetRegistry();
+		registry.define('date-cell', customDateCell);
+
+		return registry;
+	}
+
+	private _getMonthLength(month: number, year: number) {
+		const lastDate = new Date(year, month + 1, 0);
+		return lastDate.getDate();
+	}
+
+	private _getMonthYear() {
+		const {
+			month,
+			selectedDate = this._defaultDate,
+			year
+		} = this.properties;
+		return { month: typeof month === 'number' ? month : selectedDate.getMonth(), year: year ? year : selectedDate.getFullYear() };
+	}
+
+	private _goToDate(day: number) {
+		const {
+			month,
+			year
+		} = this._getMonthYear();
+		const currentMonthLength = this._getMonthLength(month, year);
+		const previousMonthLength = this._getMonthLength(month - 1, year);
+
+		if (day < 1) {
+			this._onMonthDecrement();
+			day += previousMonthLength;
+		}
+		else if (day > currentMonthLength) {
+			this._onMonthIncrement();
+			day -= currentMonthLength;
+		}
+
+		this._focusedDay = day;
+		this._callDateFocus = true;
+		this.invalidate();
+	}
+
+	private _onDateClick(date: number, disabled: boolean) {
+		const { onDateSelect } = this.properties;
+		let {
+			month,
+			year
+		} = this._getMonthYear();
+
+		if (disabled && date < 15) {
+			({ month, year } = this._onMonthIncrement());
```

Comment:
It didn't work for me otherwise, and that's the syntax I see in the MDN examples: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment